cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -404,7 +437,20 @@ public void deleteSchedules(ProgramId programId) {
         queue.markJobsForDeletion(scheduleId, deleteTime);
       }
       for (ProgramSchedule programSchedule : schedules) {
-        adminEventPublisher.publishScheduleDeletion(programSchedule.getScheduleId(), programSchedule);
+        ScheduleId scheduleId = programSchedule.getScheduleId();
+        adminEventPublisher.publishScheduleDeletion(scheduleId, programSchedule);
+        // if the deleted schedule has properties with profile assignment, remove the assignment
+        Optional<ProfileId> profileId = SystemArguments.getProfileIdFromArgs(scheduleId.getNamespaceId(),
+                                                                             programSchedule.getProperties());
+        if (profileId.isPresent()) {
+          try {
+            profileDataset.removeProfileAssignment(profileId.get(), scheduleId);
+          } catch (NotFoundException e) {
+            // this should not happen since the profile cannot be deleted if there is a schedule who is using it
+            LOG.warn("Unable to find the profile {} when deleting schedule {}, " +
+                       "therefore unable to remove the assignment", profileId.get(), scheduleId);
```

Comment:
I think we can get rid of 'therefore unable to remove the assignment', as it's functionally the same as if it were removed. Let's change it to ' delete schedule {}, skipping assignment deletion.'